types/src/v2/error.rs

```diff
@@ -21,18 +21,50 @@ pub struct JsonRpcError<'a> {
 pub struct JsonRpcErrorAlloc {
 	/// JSON-RPC version.
 	pub jsonrpc: TwoPointZero,
-	/// Error object.
-	pub error: ErrorCode,
+	/// JSON-RPC error object.
+	pub error: JsonRpcErrorObjectAlloc,
 	/// Request ID.
 	pub id: Id,
 }
 
 impl fmt::Display for JsonRpcErrorAlloc {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		write!(f, "{:?}: {}: {:?}", self.jsonrpc, self.error, self.id)
+		write!(f, "{:?}: {:?}: {:?}", self.jsonrpc, self.error, self.id)
 	}
 }
 
+/// JSON-RPC error object.
+#[derive(Debug, PartialEq, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct JsonRpcErrorObjectAlloc {
+	/// Code
+	pub code: JsonRpcErrorCode,
+	/// Message
+	pub message: String,
+	/// Optional data
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub data: Option<JsonValue>,
+}
+
+impl From<JsonRpcErrorCode> for JsonRpcErrorObjectAlloc {
+	fn from(code: JsonRpcErrorCode) -> Self {
+		Self { message: code.message().to_owned(), code, data: None }
```

Comment:
```suggestion
		Self { code, message: code.message().to_owned(), data: None }
```
…to keep clippy happy?